cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -193,8 +193,13 @@ public synchronized void create(final NamespaceMeta metadata) throws Exception {
       }
     } catch (Throwable t) {
       LOG.error(String.format("Failed to create namespace '%s'", namespace.getNamespace()), t);
-      // failed to create namespace in underlying storage so delete the namespace meta stored in the store earlier
-      deleteNamespaceMeta(metadata.getNamespaceId());
+      try {
+        resourceDeleter.get().deleteResources(metadata);
```

Comment:
need to have 2 try - catch blocks, one for this and one for deleteNamespaceMeta. Otherwise if this one throws an exception we won't try to delete the namespace meta.